gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/membership/domain_service/IntegrationPrimaryOwnerDomainService.java

```diff
@@ -54,10 +66,71 @@ public void createIntegrationPrimaryOwnerMembership(String integrationId, Primar
             });
     }
 
+    public Single<PrimaryOwnerEntity> getApiPrimaryOwner(final String organizationId, String integrationId)
```

Comment:
```suggestion
    public Maybe<PrimaryOwnerEntity> findPrimaryOwner(final String organizationId, String integrationId)
```

I would return a Maybe instead of throwing a not found exception. This function is used to build a view of the Integration if for any reason the PrimaryOwner can't be found (I guess if the DB is corrupted) it does not prevent "rendering" the response.

I'm afraid is this happens we would get a 404 response that can be misleading because the integration exists 🤔